spec/services/oai_processing_service_spec.rb

```diff
@@ -25,10 +31,27 @@
     end
 
     it 'calls the Traject command to process the xml' do
-      response = solr.get('select')
-
       expect(Traject::Indexer::MarcIndexer).to respond_to(:new)
-      expect(response['response']['numFound']).to eq 4
+      expect(number_of_docs).to eq 4
+    end
+
+    context 'reindexing' do
+      context 'ensuring non-duplication' do
+        let(:response_ids) { response['response']['docs'].map { |d| d['id'] } }
+        let(:ids_array) do
+          ["990005651670302486", "990000954720302486", "990028391040302486", "990002589250302486"]
+        end
+
+        it 'produces the same unique records when running the indexer on the same material' do
+          expect(number_of_docs).to eq 4
+          expect(response_ids).to match_array(ids_array)
+
+          process_indexer
+
+          expect(number_of_docs).to eq 4
+          expect(response_ids).to match_array(ids_array)
```

Comment:
I was thinking of the [change matcher](https://relishapp.com/rspec/rspec-expectations/v/2-0/docs/matchers/expect-change#expecting-no-change) for this. Are `number_of_docs` and `response_ids` updated after the indexer is run, or are their values cached?

Reply:
You're right, the values are cached. Making changes.

Reply:
@rotated8 Ready for a re-review.